### PR TITLE
Configuration Assessment module implementation

### PR DESCRIPTION
**Pull request by TJOSERAFAEL**
### Description

The implementation of the Configuration Assessment module is a new replacement for the old rootcheck audit functionality with great imporvements.

- The audit files are using the YAML format for better readability and structure.
- The events sended to the manager are stored in the database, allowing to visualize the data easily.
- Each agent has an internal hash table, sending only the differences to the manager.
- Manager can check for integrity with the database, requesting a full DB to an agent if data is missing.

The related issue https://github.com/wazuh/wazuh/issues/1983

### CODE REVIEW

File: src/wazuh_modules/wm_configuration_assessment.c [hunk 1]
```diff
@@ -0,0 +1,2291 @@
...
+        }
+        cJSON_AddStringToObject(check, "title", title->valuestring);
+    } else {
+        merror("No 'title' field found on check '%d'",pm_id->valueint);
+        goto error;
+    }
+
+    if(!policy_id){
+        merror("No 'id' field found on policy");
+        goto
```
Comment (chemamartinez) [inline src/wazuh_modules/wm_configuration_assessment.c]: `configuration_assessment` -> `configuration-assessment`

File: src/wazuh_modules/wm_configuration_assessment.c [hunk 2]
```diff
@@ -0,0 +1,2291 @@
...
+
+    /* Create summary hash for each policy file */
+    if(data->profile){
+        for(i = 0; data->profile[i]; i++) {
+            os_realloc(last_md5, (i + 2) * sizeof(char *), last_md5);
+            os_calloc(1,sizeof(os_md5),last_md5[i]);
+        }
+    }
+
+    while ((data->queue = StartMQ(DEFAULTQPATH, WRITE)) < 0) {
```
Comment (chemamartinez) [inline src/wazuh_modules/wm_configuration_assessment.c]: Infinite loop when the queue is not available. Set a counter here.

File: etc/templates/config/centos/5/configuration_assessment.template [hunk 3]
```diff
@@ -0,0 +1,15 @@
+  <configuration_assessment>
```
Comment (chemamartinez) [inline etc/templates/config/centos/5/configuration_assessment.template]: For every template:
- `configuration_assessment` -> `configuration-assessment`
- Remove application policies from the templates (Apache & MySQL)

**Comment by chemamartinez** [comment]
New rootcheck templates should not contain options related to Policy monitoring.

File: etc/rules/0570-policy_rules.xml [hunk 4]
```diff
@@ -10,42 +10,79 @@
...
   </rule>
 
-  <rule id="19004" level="3">
-    <if_sid>19002</if_sid>
+  <rule id="19008" level="3">
+    <if_sid>19006</if_sid>
     <field name="configuration_assessment.check.result">^passed</field>
     <options>no_full_log</options>
-    <description>Check with id $(configuration_assessment.check.id) for policy $(configuration_assessment.policy) passed.</description>
+    <description>$(configuration_assessment.name): $(configuration_assessment.check.title): Passed result.</description>
```
Comment (snaow) [inline etc/rules/0570-policy_rules.xml]: Take out the "pass/failed result" keyword at the end.
`$(configuration_assessment.name): $(configuration_assessment.check.title): Passed result.`
->
`$(configuration_assessment.name): $(configuration_assessment.check.title)`

The rule descripcion must be only the check name/title.
The result/status of the check is already present on the field `check.result`.

File: etc/rules/0570-policy_rules.xml [hunk 5]
```diff
@@ -10,42 +10,79 @@
...
   </rule>
 
-  <rule id="19005" level="3">
-    <if_sid>19004</if_sid>
+  <rule id="19009" level="3">
+    <if_sid>19008</if_sid>
     <field name="configuration_assessment.check.previous_result">^failed</field>
     <options>no_full_log</options>
-    <description>Check with id $(configuration_assessment.check.id) for policy $(configuration_assessment.policy) has changed to passed.</description>
+    <description>$(configuration_assessment.name): $(configuration_assessment.check.title): Result changed from failed to passed.</description>
```
Comment (snaow) [inline etc/rules/0570-policy_rules.xml]: `$(configuration_assessment.name): $(configuration_assessment.check.title): Result changed from failed to passed.`
->
`$(configuration_assessment.name): $(configuration_assessment.check.title): Status changed from failed to passed`

File: etc/rules/0570-policy_rules.xml [hunk 6]
```diff
@@ -10,42 +10,79 @@
...
   </rule>
 
-  <rule id="19006" level="9">
-    <if_sid>19003</if_sid>
+  <rule id="19010" level="9">
+    <if_sid>19007</if_sid>
     <field name="configuration_assessment.check.previous_result">^passed</field>
     <options>no_full_log</options>
-    <description>Check with id $(configuration_assessment.check.id) for policy $(configuration_assessment.policy) has changed to failed.</description>
+    <description>$(configuration_assessment.name): $(configuration_assessment.check.title): Result changed from passed to failed.</description>
```
Comment (snaow) [inline etc/rules/0570-policy_rules.xml]: `$(configuration_assessment.name): $(configuration_assessment.check.title): Result changed from passed to failed.`
->
`$(configuration_assessment.name): $(configuration_assessment.check.title): Status changed from passed to failed.`

**Pull request merged**